lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -3648,6 +3648,55 @@ class DecomposeAtenStackOp : public OpRewritePattern<AtenStackOp> {
 };
 } // namespace
 
+// Decompose `aten.hstack` into `aten.at_least1d` and `aten.cat`.
+// https://github.com/pytorch/pytorch/blob/207564bab1c4fe42750931765734ee604032fb69/torch/_refs/__init__.py#L3908
+namespace {
+class DecomposeAtenHstackOp : public OpRewritePattern<AtenHstackOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenHstackOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+
+    // Get SmallVector<Value> from Value.
+    SmallVector<Value> tensors;
+    if (!getListConstructElements(op.getTensors(), tensors))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: the tensor list is not from list construct");
+
+    // Create constants.
+    auto zero =
+        rewriter.create<ConstantIntOp>(loc, rewriter.getI64IntegerAttr(0));
+    auto one =
+        rewriter.create<ConstantIntOp>(loc, rewriter.getI64IntegerAttr(1));
+
+    // Execute AtenAtleast1dOp on every tensor inside tensors.
+    SmallVector<Value> atleast1dTensors;
+    for (auto tensor : tensors) {
+      auto atleast1dTensor =
+          rewriter.create<AtenAtleast1dOp>(loc, tensor.getType(), tensor);
+      atleast1dTensors.push_back(atleast1dTensor);
+    }
```

Comment:
Before doing this, shouldn't we check whether the tensors are already of rank >= 1 or not? Otherwise, this would result in a lot of unusable code.